File: 1221-element-appearing-more-than-25-in-sorted-array/element-appearing-more-than-25-in-sorted-array.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        n = len(arr)
        map = {}
        for num in arr:
            if num in map:
                map[num] = map[num] + 1
            else:
                map[num] = 1
            if map[num] > n // 4:
                return num
        return -1
        

File: 1221-element-appearing-more-than-25-in-sorted-array/element-appearing-more-than-25-in-sorted-array.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
       unordered_map<int,int>map;
       int n = arr.size();
       for(int i = 0; i < n; i++){
           map[arr[i]]++;
           if(map[arr[i]] > n / 4) return arr[i];
       }
       return -1;
    }
};